test/src/utils/vectors/vector_basic.test.cpp

```diff
@@ -29,6 +28,17 @@ namespace utils
 {
 namespace math
 {
+
+std::array<int, 3> createRandomArrayForBasic()
+{
+  std::srand(time(0));
```

Comment:
Do we really want to do the seeding every time we run this function?

I don't know how expensive it is, but doesn't it make more sense to do it in some setup/init function?

Reply:
@pmorande27 any thoughts on this?

Reply:
If we don't do this then it will always generate the same "random" number, each time that you call it in your machine. While it is true that it has some cost I think it is worth it for coverage. Each time that you commit the test will be different

Reply:
If we choose to remove it will generate a truly random number in the google actions part, but I think it is more helpful to have a real random number in the commit part, so that if you have modified something and it goes wrong then it won't let you commit

Reply:
I'm not denying that. All I'm saying is that it should be sufficient to do this *once* after starting the program, not every time we need another number (or set thereof).

Read more [here](https://en.cppreference.com/w/cpp/numeric/random/srand).

In particular ...
> If `rand()` is used before any call to `srand()`, `rand()` behaves as if it was seeded with `srand(1)`.

... indicating that it's required, and

> Each time `rand()` is seeded with the same seed, it must produce the same *sequence* of values.

... showing that seeding once is indeed sufficient because while the sequence is predefined, there is no real correlation between the values of that sequence. So using the time to seed on startup will yield a new sequence every time.

Reply:
Besides performance, it's even better to only seed once because it allows us to get the seed used and fix it if we want to analyse a particular case! (As one seed defines the entire program behaviour.)

Reply:
Okay! We can try to have a look into it to figure it out :)

Reply:
So would the suggestion be to change the generation of the random into the Test structs and save it as a variable in there for the tests to use and if we seed it with the time it will always be different every time we run the tests?

Reply:
I'm not sure I understand, but I want to say no.

Essentially, we want to make sure that `std::srand(time(0))` is called *once* during each execution (each time we invoke `make test` or similar) but before the first call to `rand()`. I would put it in the `SetUp` function. You don't need to save anything anywhere, I think. 

What your doing at the moment totally works in terms of correctness, but I think it's too much. You don't need to call it every time you generate a new random vector.